qiskit_nature/problems/sampling/protein_folding/interactions/random_interaction.py

```diff
@@ -0,0 +1,33 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A class defining a random interaction between beads of a peptide."""
+from typing import List
+
+import numpy as np
+from .interaction import Interaction
+
+
+class RandomInteraction(Interaction):
+    """A class defining a random interaction between beads of a peptide."""
+
+    def calc_energy_matrix(self, chain_len: int, residue_sequence: List[str] = None) -> np.ndarray:
+        """
+        Calculates an energy matrix for a random interaction.
+        Args:
+            chain_len: Length of a protein chain.
+            residue_sequence: None
+
+        Returns:
+            pair_energies: Numpy array of pair energies for amino acids.
+        """
+        pair_energies = -1 - 4 * np.random.rand(chain_len + 1, 2, chain_len + 1, 2)
```

Comment:
Since I am seeing randomness here, I quickly need to ask whether we are dealing with seeds properly in Nature.. Also tagging @woodsp-ibm @pbark

Reply:
I am not sure what you mean with this @mrossinek . Is this related to unit tests on random cases where we need to fix the seed? Here we want to introduce random interactions for contact maps of proteins, so fixing the seed would not be the way to go. If this is more general about how we treat random occasions in qiskit-nature we should have a separate discussion.

Reply:
It might be better to use the random number generator we have for algorithms in Terra. We have done this elsewhere and means that if you want to have a reproducible outcome for testing or whatever you just have the one seed to set. In normal operation the seed is chosen randomly so things are random; but you can manually seed for reproducible results like for unit tests, tutorials where you want a predictable outcome etc.

Reply:
It may also be important for normal operations. Consider an example where you had a random seed and something unexpected occurred. You could then check the logs, find the seed, and be able to reproduce and investigate the unexpected behavior. As it is now, this will not be possible.

Reply:
I agree with Steve. This needs to be addressed.